src/hotspot/os_cpu/bsd_aarch64/os_bsd_aarch64.cpp

```diff
@@ -0,0 +1,737 @@
+               sig == SIGBUS && /* info->si_code == BUS_OBJERR && */
+               thread->doing_unsafe_access()) {
+      address next_pc = pc + NativeCall::instruction_size;
+      if (UnsafeCopyMemory::contains_pc(pc)) {
+        next_pc = UnsafeCopyMemory::page_error_continue_pc(pc);
+      }
+      stub = SharedRuntime::handle_unsafe_access(thread, next_pc);
+    }
+
+    // jni_fast_Get<Primitive>Field can trap at certain pc's if a GC kicks in
+    // and the heap gets shrunk before the field access.
+    if ((sig == SIGSEGV) || (sig == SIGBUS)) {
+      address addr = JNI_FastGetField::find_slowcase_pc(pc);
+      if (addr != (address)-1) {
+        stub = addr;
+      }
+    }
+  }
+
+#if defined(ASSERT) && defined(__APPLE__)
+  // Execution protection violation
+  //
+  // This should be kept as the last step in the triage.  We don't
+  // have a dedicated trap number for a no-execute fault, so be
+  // conservative and allow other handlers the first shot.
+  if (UnguardOnExecutionViolation > 0 &&
+      (sig == SIGBUS)) {
+    static __thread address last_addr = (address) -1;
+
+    address addr = (address) info->si_addr;
+    address pc = os::Posix::ucontext_get_pc(uc);
+
+    if (pc != addr && uc->context_esr == 0x9200004F) { //TODO: figure out what this value means
+      // We are faulting trying to write a R-X page
+      pthread_jit_write_protect_np(false);
+
+      log_debug(os)("Writing protection violation "
+                    "at " INTPTR_FORMAT
+                    ", unprotecting", p2i(addr));
+
+      stub = pc;
+
+      last_addr = (address) -1;
+    } else if (pc == addr && uc->context_esr == 0x8200000f) { //TODO: figure out what this value means
+      // We are faulting trying to execute a RW- page
+
+      if (addr != last_addr) {
+        pthread_jit_write_protect_np(true);
+
+        log_debug(os)("Execution protection violation "
+                      "at " INTPTR_FORMAT
+                      ", protecting", p2i(addr));
+
+        stub = pc;
+
+        // Set last_addr so if we fault again at the same address, we don't end
+        // up in an endless loop.
+        last_addr = addr;
+      }
+    }
+  }
+#endif
+
+  if (stub != NULL) {
+    // save all thread context in case we need to restore it
+    if (thread != NULL) thread->set_saved_exception_pc(pc);
+
+    os::Posix::ucontext_set_pc(uc, stub);
+    return true;
+  }
+
+  return false; // Mute compiler
+}
+
+void os::Bsd::init_thread_fpu_state(void) {
+}
+
+bool os::is_allocatable(size_t bytes) {
+  return true;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// thread stack
+
+// Minimum usable stack sizes required to get to user code. Space for
+// HotSpot guard pages is added later.
+size_t os::Posix::_compiler_thread_min_stack_allowed = 72 * K;
+size_t os::Posix::_java_thread_min_stack_allowed = 72 * K;
+size_t os::Posix::_vm_internal_thread_min_stack_allowed = 72 * K;
```

Comment:
Those are slightly larger than their x86_64 counter parts. Are they conservative/aggressive values? How did we arrive at those?

Reply:
These values were copied from linux_aarch64. The motivation is that clang on macos/aarch64 will likely to produce stack frames for C++ functions similar to frames generates by gcc on linux/aarch64. And sizes of java stack frames should not change.